### PR TITLE
Fix train code of Faster R-CNN to properly handle labels in range [0, n_fg_class - 1]

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

File: chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py [hunk 1]
```diff
@@ -75,7 +75,9 @@ def __call__(self, imgs, bboxes, labels, scale):
             bboxes (~chainer.Variable): A batch of bounding boxes.
                 Its shape is :math:`(N, R, 4)`.
             labels (~chainer.Variable): A batch of labels.
-                Its shape is :math:`(N, R)`.
+                Its shape is :math:`(N, R)`. The background is excluded from
+                the definition, which means that the range of the value
+                is :obj:`[0, self.faster_rcnn.n_class - 2]`.
```
Comment (Hakuyume) [inline chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py]: `:math:[0, self.faster_rcnn.n_class - 2]`

Comment (yuyu2172) [inline chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py]: I don't like to use `self` inside `:math`.
How about the change I made?

**Comment by Hakuyume** [comment]
How about adding a note about the range of the input of `ProposalTargetCreator`?
https://github.com/pfnet/chainercv/blob/master/chainercv/links/model/faster_rcnn/utils/proposal_target_creator.py#L72

File: chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py (hunk 1 above)
Comment (Hakuyume) [inline chainercv/links/model/faster_rcnn/faster_rcnn_train_chain.py]: I think that is good.

File: chainercv/links/model/faster_rcnn/utils/proposal_target_creator.py [hunk 2]
```diff
@@ -85,14 +86,19 @@ def __call__(self, roi, bbox, label,
...
 
         """
         xp = cuda.get_array_module(roi)
         roi = cuda.to_cpu(roi)
         bbox = cuda.to_cpu(bbox)
         label = cuda.to_cpu(label)
 
+        # Offset range of classes from [0, n_class - 2] to [1, n_class - 1].
+        # The label with value 0 is the background.
+        label = label + 1
```
Comment (Hakuyume) [inline chainercv/links/model/faster_rcnn/utils/proposal_target_creator.py]: How about adding `1` here
```
gt_roi_label = label[gt_assignment] + 1
```

Comment (yuyu2172) [inline chainercv/links/model/faster_rcnn/utils/proposal_target_creator.py]: Thanks. I also fixed a comment to use `n_fg_class`.

**Comment by Hakuyume** [comment]
LGTM

**Pull request merged**